### ISSUE TITLE
only use int in src

### ISSUE THREAD

**Issue by mpadge** [ropensci/osmdata#124]
The [google C++ style guide](https://google.github.io/styleguide/cppguide.html#Integer_Types) is a good argument against usage of `unsigned int` and the like. Modify all `src/` code to follow this.

**Comment by Robinlovelace** [comment]
Just read this and the previous issue. Beyond my Ken but I'm imagine Andrew Smith may be interested. Any comments on suggestions @virgesmith?

**Comment by virgesmith** [comment]
Personally I think google C++ style guide is more a "how to live with millions of lines of legacy C++ and survive" (as google do) rather than a modern C++ best practice. I would argue that `long` is more of a concern than unsigned as it's size is platform-dependent and this can give you grief porting between 64 bit windows (LLP64) and linux/osx (LP64)

Signedness is a problem - unfortunately thanks to STL you cant really get rid of it internally without having static_cast all over the place. Try doing it for a ~3.5M line codebase - we tried, we gave up). But see this ancient scroll by Scott Meyers: http://www.aristeia.com/Papers/C++ReportColumns/sep95.pdf

PS Personally I rather like writing `uint32_t x = -1u; ` - -1 is easier to remember than 4billion and something ;)
